docs/usecase.rst

```diff
@@ -37,14 +72,37 @@ compute/data/communications abstractions,
 quality of service guarantees,
 and performance optimizations.
 
+.. seealso:: :doc:`executor`
+
 Generic sequential task execution
 ---------------------------------
 
 Units of work (nodes) describe the inputs and expected outputs,
 and identify a module able to perform the operation.
 
+| 1. Receive a node of work to execute.
+| 2. Import a module implementing the operation.
+| 3. Initialize or restore checkpoint state.
+| 4. Prepare operation resources.
+|    3.1 Request or confirm locality of input data.
+|    3.2 Create data publishing resource for operation.
+|    3.3 Prepare compute environment for execution.
+|    3.4 Use operation module helpers to construct input objects.
```

Comment:
I think I will need more details to properly understand this. The use of 'operation module helpers' and 'input objects' is specific to the gmxapi context?

Reply:
It is not intended to be specific to gmxapi; there is a C++ implementation that is specific to the GROMACS MD code, but the pattern is implemented in pure Python, as well, with duck typing instead of package dependencies. Prototypes exist in various stages for C++ interoperability of operation implementations and executors through template headers and/or PyCapsule conventions. It is intended to be part of a generic and extensible framework by which operations and executors may be developed independently with varying facilities, requirements, and levels of integration.

Reply:
Thanks. Can we have pointers to the C++ and Python code you mentioned?

Reply:
> Thanks. Can we have pointers to the C++ and Python code you mentioned?

* https://github.com/eirrgang/gmxapi/blob/75dd25605193a307e2112f80f31442e22e6d9fa5/python_packaging/sample_restraint/src/pythonmodule/export_plugin.cpp#L47
* https://github.com/eirrgang/gmxapi/blob/894531636b81554c6eea72593c32476d5511a250/python_packaging/src/gmxapi/simulation/context.py#L837
* https://kassonlab.github.io/gmxapi/userguide/pythonreference.html#gmxapi.function_wrapper
* https://github.com/kassonlab/gmxapi/blob/894531636b81554c6eea72593c32476d5511a250/python_packaging/src/gmxapi/operation.py#L1935
* https://github.com/kassonlab/gmxapi/blob/master/python_packaging/src/gmxapi/simulation/modify_input.py

Reply:
Also relevant:
* https://github.com/kassonlab/gmxapi/blob/894531636b81554c6eea72593c32476d5511a250/python_packaging/src/gmxapi/operation.py#L1164
* https://github.com/kassonlab/gmxapi/blob/894531636b81554c6eea72593c32476d5511a250/python_packaging/src/gmxapi/abc.py#L464